src/capabilities.ts

```diff
@@ -0,0 +1,64 @@
+import semver from 'semver';
+import buildTargets from './buildTargets';
+
+/**
+ * Device capabilities supported by the application being built will not be
+ * included into the manifest when targeting a JS API version lower than this.
+ */
+const includeSupportedDeviceCapsSince = '7.0.0';
+
+/**
+ * Names for known device capabilities.
+ */
+export enum DeviceCapability {
+  HW_SCREEN_SIZE = 'hw.screen.size',
```

Comment:
nit: Should this perhaps be `{ hw: { screen: { size: [] } } }`?

Reply:
Dropped nested structs entirely to `screenSize: { w: number, h: number }`.